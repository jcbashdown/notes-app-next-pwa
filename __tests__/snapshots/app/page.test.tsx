import { render, act } from '@testing-library/react'
import Home from '@/app/page'

it('renders homepage unchanged', async () => {
    const { container } = await act(async () => render(<Home />))
    expect(container).toMatchSnapshot()
})
