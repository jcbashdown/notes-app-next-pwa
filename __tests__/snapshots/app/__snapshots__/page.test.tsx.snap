// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-100 p-4 flex flex-col items-center"
  >
    <div
      class="w-full max-w-3xl"
    >
      <div
        class="bg-white p-4 rounded-lg shadow-md mb-4 flex"
      >
        <input
          class="w-full p-2 border border-gray-300 rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-400"
          data-id="newTopic"
          placeholder="Add a new note..."
          type="text"
          value=""
        />
        <div>
          <button
            class="ml-2 p-2 border border-gray-300 rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-400"
            id="menuButton"
          >
            <svg
              class="w-6 h-6 text-gray-700"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="hidden absolute right-0 mt-2 w-48 bg-white border border-gray-200 rounded-lg shadow-lg"
            id="menuDropdown"
          >
            <a
              class="block px-4 py-2 text-gray-800 hover:bg-gray-200"
              href="#"
              id="download-note-json"
            >
              Download notes
            </a>
          </div>
        </div>
      </div>
      <div
        class="space-y-4"
      >
        <div
          class="bg-white p-4 rounded-lg shadow-md"
        >
          <input
            data-id="f2fb0a80-9f6a-4b5c-8b4c-c5e5e5d5f5d5_topic"
            type="text"
            value="This is the root note"
          />
          <ul
            class="ml-2"
          >
            <li>
              <span
                class="p-1"
              >
                <input
                  class="pl-1"
                  data-id="f2fb0a80-9f6a-4b5c-8b4c-c5e5e5d5f5d5---a1b2c3d4-e5f6-4b5c-8b4c-c5e5e5d5f5d5_relationship"
                  type="text"
                  value="+"
                />
                <input
                  data-id="f2fb0a80-9f6a-4b5c-8b4c-c5e5e5d5f5d5---a1b2c3d4-e5f6-4b5c-8b4c-c5e5e5d5f5d5_text"
                  type="text"
                  value="This note supports the root note"
                />
              </span>
              <ul
                class="ml-2"
              >
                <li>
                  <span
                    class="p-1"
                  >
                    <input
                      class="pl-1"
                      data-id="a1b2c3d4-e5f6-4b5c-8b4c-c5e5e5d5f5d5---a0b1c2d3-e4f5-4b5c-8b4c-c5e5e5d5f5d5_relationship"
                      type="text"
                      value=""
                    />
                    <input
                      data-id="a1b2c3d4-e5f6-4b5c-8b4c-c5e5e5d5f5d5---a0b1c2d3-e4f5-4b5c-8b4c-c5e5e5d5f5d5_text"
                      type="text"
                      value="This note is related to note 2"
                    />
                  </span>
                  <ul
                    class="ml-2"
                  />
                </li>
              </ul>
            </li>
            <li>
              <span
                class="p-1"
              >
                <input
                  class="pl-1"
                  data-id="f2fb0a80-9f6a-4b5c-8b4c-c5e5e5d5f5d5---1a2b3c4d-5e6f-4b5c-8b4c-c5e5e5d5f5d5_relationship"
                  type="text"
                  value="-"
                />
                <input
                  data-id="f2fb0a80-9f6a-4b5c-8b4c-c5e5e5d5f5d5---1a2b3c4d-5e6f-4b5c-8b4c-c5e5e5d5f5d5_text"
                  type="text"
                  value="This note opposes the root note"
                />
              </span>
              <ul
                class="ml-2"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
